From 7580a360774a1514b6177b4049cab9eb4d8c35bc Mon Sep 17 00:00:00 2001
From: Huaxiang Sun <hsun@cloudera.com>
Date: Mon, 16 Nov 2015 10:36:44 -0800
Subject: [PATCH 280/295] Connection Idle time 1 second is too short and the
 connection is closed too quickly by the
 ChoreService. Increase it to the default (10
 minutes) for testAll(). The patch for HBASE-14533
 is not committed to upstream yet, so this fix
 cannot be mapped to upstream.

Reason: Bug
Author: Huaxiang Sun
Ref: CDH-33980

Change-Id: I52fc378e18537317b036549b6b4bd46f014da28b
---
 .../hadoop/hbase/thrift/TestThriftServer.java      |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/hbase-thrift/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServer.java b/hbase-thrift/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServer.java
index df42403..5707cf8 100644
--- a/hbase-thrift/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServer.java
+++ b/hbase-thrift/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServer.java
@@ -100,10 +100,6 @@ public class TestThriftServer {
 
   @BeforeClass
   public static void beforeClass() throws Exception {
-    UTIL.getConfiguration().setInt(ThriftServerRunner.HBaseHandler.MAX_IDLETIME,
-        CONNECTION_IDLE_TIME);
-    UTIL.getConfiguration().setInt(ThriftServerRunner.HBaseHandler.CLEANUP_INTERVAL,
-        RUN_CLEANER_INTERVAL);
     UTIL.getConfiguration().setBoolean(ThriftServerRunner.COALESCE_INC_KEY, true);
     UTIL.getConfiguration().setBoolean("hbase.table.sanity.checks", false);
     UTIL.startMiniCluster();
@@ -117,9 +113,15 @@ public class TestThriftServer {
   @Test (timeout=30000)
   public void testConnectionCache()
   throws IOException, IOError, IllegalArgument, AlreadyExists, InterruptedException {
+    Configuration conf = new Configuration(UTIL.getConfiguration());
+    conf.setInt(ThriftServerRunner.HBaseHandler.MAX_IDLETIME,
+        CONNECTION_IDLE_TIME);
+    conf.setInt(ThriftServerRunner.HBaseHandler.CLEANUP_INTERVAL,
+        RUN_CLEANER_INTERVAL);
+
     ThriftServerRunner.HBaseHandler handler =
-        new ThriftServerRunner.HBaseHandler(UTIL.getConfiguration(),
-            UserProvider.instantiate(UTIL.getConfiguration()));
+        new ThriftServerRunner.HBaseHandler(conf,
+            UserProvider.instantiate(conf));
     final ByteBuffer tn = asByteBuffer("tn");
     handler.createTable(tn, getColumnDescriptors());
     List<Mutation> mutations = new ArrayList<Mutation>(1);
-- 
1.7.9.5

